src/App.tsx

```diff
@@ -15,47 +16,115 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((a, b) => {
+    switch (sortType) {
+      case 1:
+        return a.localeCompare(b);
+
+      case 2:
+        return a.length - b.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, handleReverse] = useState(false);
+  const [sortType, changeSortType] = useState(0);
+
+  const handleReset = () => {
+    changeSortType(0);
+    handleReverse(false);
+  };
+
+  const goods = getReorderedGoods(goodsFromServer, { isReversed, sortType });
+  const resetButtonVisible = isReversed || sortType !== 0;
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={
+            classNames('button is-info', {
+              'is-light': sortType !== 1,
```

Comment:
it will be more readable if write not number but what it mean in enum:
```suggestion
              'is-light': sortType !== SortTyepe. ALPHABET,
```